cmd/clusterctl/cmd/delete.go

```diff
@@ -17,15 +17,82 @@ limitations under the License.
 package cmd
 
 import (
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
+type deleteOptions struct {
+	kubeconfig           string
+	forceDeleteNamespace bool
+	forceDeleteCRD       bool
+	deleteAll            bool
+}
+
+var dd = &deleteOptions{}
+
 var deleteCmd = &cobra.Command{
-	Use:   "delete",
-	Short: "Delete a cluster API resource",
-	Long:  `Delete a cluster API resource with one command`,
+	Use:   "delete [providers]",
+	Short: "Deletes one or more providers from the management cluster",
+	Long: LongDesc(`
+		Deletes one or more providers from the management cluster.`),
+
+	Example: Examples(`
+		# Deletes the AWS provider
+		# Please note that this imply the deletion of everything all the provider components except the 
+		# hosting namespace and the CRDs.
+		clusterctl delete aws
+ 
+		# Deletes the instance of the AWS provider hosted in the "foo" namespace
+		# Please note that, in case there are more than one instances of the AWS provider installed in the cluster,
+		# the provider components shared across providers (e.g. ClusterRoles), are not deleted in order to preserve
+		# the functioning of the remaining instances.
+		clusterctl delete foo/aws
```

Comment:
I think this is counter intuitive to what most people are used to. Typically it's "delete kind/resource name", it's  kinda odd to see namespace there. Wondering if it's better to add a `--namespace` flag instead?

Reply:
Ok, fair point.
I switched to the flag, it is used only here and it isn't a big deal